pkg/storage/bucket/s3/bucket_client.go

```diff
@@ -92,6 +93,7 @@ func newS3Config(cfg Config) (s3.Config, error) {
 }
 
 type BucketWithRetries struct {
+	logger           log.Logger
```

Comment:
Do we need to pass the logger when intializing `BucketWithRetries`? I don't see that change in this pr.